client/state/data-layer/wpcom/account-recovery/lookup/index.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import { isString, tap } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import { ACCOUNT_RECOVERY_RESET_OPTIONS_REQUEST } from 'state/action-types';
+import {
+	fetchResetOptionsSuccess,
+	fetchResetOptionsError,
+} from 'state/account-recovery/reset/actions';
+
+export const fromApi = data => ( [
+	{
+		email: data.primary_email,
+		sms: data.primary_sms,
+		name: 'primary',
+	},
+	{
+		email: data.secondary_email,
+		sms: data.secondary_sms,
+		name: 'secondary',
+	},
+] );
+
+export const validate = ( { primary_email, primary_sms, secondary_email, secondary_sms } ) => {
+	if ( ! [ primary_email, primary_sms, secondary_email, secondary_sms ].every( isString ) ) {
+		throw Error( 'Unexpected response format from /account-recovery/lookup' );
+	}
+};
+
+export const handleRequestResetOptions = ( { dispatch }, { userData } ) => (
+	wpcom.req.get( {
+		body: userData,
+		apiNamespace: 'wpcom/v2',
+		path: '/account-recovery/lookup',
+	} ).then( data => dispatch( fetchResetOptionsSuccess( fromApi( tap( data, validate ) ) ) ) )
+	.catch( error => dispatch( fetchResetOptionsError( error ) ) )
```

Comment:
That's totally true : ) However, I thought the [apiNamespace issue](https://github.com/Automattic/wp-calypso/pull/10878#issuecomment-282268167) was still there? Thus, besides the difficulty of rebasing, part of the reason that I opened this new PR was to rewrite it using the old `wpcom.req` way, because the previous #10878 was already in `wpcom-http` and I had a hard time to figure out how to fix the issue. 

Maybe it has already been fixed and I have overlooked?